SoomlaAndroidLevelUp/src/com/soomla/levelup/LevelUp.java

```diff
@@ -171,12 +171,13 @@ private static void clearCurrentState() {
 
     private static void addWorldObjectToWorlds(HashMap<String, JSONObject> worlds, JSONObject worldJSON) throws JSONException {
         String worldId = worldJSON.getString("itemId");
-        worlds.put(worldId, worldJSON);
+        HashMap<String, JSONObject> worldsInternal = new HashMap<String, JSONObject>(worlds);
```

Comment:
@EugeneButusov That's not what I meant :)
I meant that only the list at line 209(210) should be cloned because it's probably the cause of the exception.